src/main/kotlin/apply/application/EvaluationService.kt

```diff
@@ -0,0 +1,96 @@
+package apply.application
+
+import apply.domain.evaluation.Evaluation
+import apply.domain.evaluation.EvaluationRepository
+import apply.domain.evaluation.dto.EvaluationResponse
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import java.util.Optional
+import javax.annotation.PostConstruct
+import javax.transaction.Transactional
+import kotlin.streams.toList
+
+@Transactional
+@Service
+class EvaluationService(
+    private val evaluationRepository: EvaluationRepository,
+    private val recruitmentService: RecruitmentService
+) {
+    private fun findAll(): List<Evaluation> {
+        return evaluationRepository.findAll()
+    }
+
+    private fun findById(id: Long): Evaluation? {
+        if (id == 0L) return null
+
+        return evaluationRepository.findByIdOrNull(id) ?: throw IllegalArgumentException("해당 id의 평가를 찾을 수 없습니다.")
+    }
+
+    fun findAllWithRecruitment(): List<EvaluationResponse> {
+        return findAll().stream()
+            .map {
+                EvaluationResponse(
+                    it.id,
+                    it.title,
+                    it.description,
+                    recruitmentService.getById(it.recruitment).title,
+                    it.recruitment,
+                    findById(it.beforeEvaluation)?.title ?: "이전 평가 없음",
+                    it.beforeEvaluation
+                )
+            }.toList()
+    }
+
+    fun deleteById(id: Long) {
+        evaluationRepository.deleteById(id)
+
+        resetBeforeEvaluationContain(id)
+    }
+
+    private fun resetBeforeEvaluationContain(id: Long) {
+        val persistEvaluations: List<Evaluation> = evaluationRepository.findAll()
+        val evaluation: Optional<Evaluation> = persistEvaluations.stream()
+            .filter { it.hasSameBeforeEvaluationWith(id) }
+            .findFirst()
+
+        if (evaluation.isPresent) {
+            evaluation.get()
+                .resetBeforeEvaluation()
+        }
```

Comment:
```suggestion
        persistEvaluations.firstOrNull { it.hasSameBeforeEvaluationWith(id) }?.resetBeforeEvaluation()
```

코틀린은 이렇게도 할 수 있어요!
자바에서 쓰던 Stream, Optional을 대체할 기능들이 존재해요~

Reply:
firstOrNull 좋네요😀
분명 써봤던 기억이 있는데 물론 까먹어버렸군요!

핀 덕분에 적용해서 코드 길이를 줄이고 가독성을 올렸습니다.
코틀린 공부를 더 열심히 해봐야겠어요!

떄떙큐